lib/ex_format/state.ex

```diff
@@ -0,0 +1,29 @@
+defmodule ExFormat.State do
```

Comment:
This should be interesting https://hexdocs.pm/elixir/Kernel.SpecialForms.html#%2525/2, but TL;DR is it checks you are using the right fields and not using unknown fields literals.